src/api/storage.ts

```diff
@@ -27,49 +25,129 @@ const writeStorageInvalid = (key: string, INITIAL_VALUE?: any) => {
+      name: 'Horário 1',
+      course_options: [],
+    },
+    {
+      id: 1,
+      icon: 'https://cdn.jsdelivr.net/npm/emoji-datasource-apple/img/apple/64/1f929.png',
+      name: 'Horário 2',
+      course_options: [],
+    },
+    {
+      id: 2,
+      icon: 'https://cdn.jsdelivr.net/npm/emoji-datasource-apple/img/apple/64/1f973.png',
+      name: 'Horário 3',
+      course_options: [],
+    },
+    {
+      id: 3,
+      icon: 'https://cdn.jsdelivr.net/npm/emoji-datasource-apple/img/apple/64/1f9d0.png',
+      name: 'Horário 4',
+      course_options: [],
+    },
+    {
+      id: 4,
+      icon: 'https://cdn.jsdelivr.net/npm/emoji-datasource-apple/img/apple/64/1f525.png',
+      name: 'Horário 5',
+      course_options: [],
+    },
+    {
+      id: 5,
+      icon: 'https://cdn.jsdelivr.net/npm/emoji-datasource-apple/img/apple/64/1f483.png',
+      name: 'Horário 6',
+      course_options: [],
+    },
+    {
+      id: 6,
+      icon: 'https://cdn.jsdelivr.net/npm/emoji-datasource-apple/img/apple/64/1f976.png',
+      name: 'Horário 7',
+      course_options: [],
+    },
+    {
+      id: 7,
+      icon: 'https://cdn.jsdelivr.net/npm/emoji-datasource-apple/img/apple/64/1f47b.png',
+      name: 'Horário 8',
+      course_options: [],
+    },
+    {
+      id: 8,
+      icon: 'https://cdn.jsdelivr.net/npm/emoji-datasource-apple/img/apple/64/1f425.png',
+      name: 'Horário 9',
+      course_options: [],
+    },
+    {
+      id: 9,
+      icon: 'https://cdn.jsdelivr.net/npm/emoji-datasource-apple/img/apple/64/1fae1.png',
+      name: 'Horário 10',
+      course_options: [],
+    },
+  ];
+
+  try {
+    if (isStorageValid(key, 7)) {
+      const multipleOptions: MultipleOptions = JSON.parse(localStorage.getItem(key))
+
+      // TODO (Process-ing): Test if this check is really necessary
+      //for (let i = 0; i < courseOptions.options.length; i++) {
+      //         for (let j = 0; j < courseOptions.options[i].length; j++) {
+      //           if (courseOptions.options[i][j].teachers === undefined) {
+      //             courseOptions.options[i][j].teachers = getCourseTeachers(courseOptions.options[i][j])
+      //           }
+      //           if (courseOptions.options[i][j].filteredTeachers === undefined) {
+      //             courseOptions.options[i][j].filteredTeachers = getCourseTeachers(courseOptions.options[i][j])
+      //           }
+      //         }
+      //       }
+      
+      //       for (let i = 0; i < courseOptions.selected.length; i++) {
+      //           if (courseOptions.selected[i].teachers === undefined) {
+      //             courseOptions.selected[i].teachers = getCourseTeachers(courseOptions.selected[i])
+      //           }
+      //           if (courseOptions.selected[i].filteredTeachers === undefined) {
+      //             courseOptions.selected[i].filteredTeachers = getCourseTeachers(courseOptions.selected[i])
+      //           }
```

Comment:
With the current logic employed by the teacher filters logic, I'm not seeing a situation where the `filteredTeachers` could be undefined, since they are always initialized with an empty array and also due to the fact that the type `CourseOption` does not allow `filteredTeachers` to be null.

Having said that, I think that the check is not necessary as I also tested with this commented and didn't catch any issues.